### ISSUE TITLE
Categorising response as Successful results in blank screen

### ISSUE THREAD

**Issue by kingqueen3065** [mysociety/alaveteli#3922]
Having made a batch request, I am categorising a very lot of requests at the moment and I'm experiencing problems when I categorise as Successful. 

I went to https://www.whatdotheyknow.com/request/lists_of_accessible_taxis_under_56 and selected
![ss 2017-04-20 at 10 35 13](https://cloud.githubusercontent.com/assets/11923618/25253644/b273e4ea-2619-11e7-9167-b083bdfa1331.png)
On "Submit Status" I am redirected to the blank page:
https://www.whatdotheyknow.com/request/401902/describe (quite literally blank - no HTML returned at all)
On reloading the blank page, I get redirected back to https://www.whatdotheyknow.com/request/lists_of_accessible_taxis_under_56 which has been categorised correctly but with the additional erroneous message "The request has been updated since you originally loaded this page. Please check for any new incoming messages below, and try again."
![ss 2017-04-20 at 10 37 24](https://cloud.githubusercontent.com/assets/11923618/25253712/fef545ac-2619-11e7-8f9c-68771f8cb774.png)
This appears to only happen on categorising the request as "successful". Categorising it as "still awaiting a response" does not result in the same error.
NB: a user complained recently that they had somehow ended up tweeting the "/describe" link for one of their requests; I couldn't work out how they'd got to such a link. I now think I do know: it's the above bug.

**Comment by kingqueen3065** [comment]
brief video demo of bug in firefox and chrome
https://www.youtube.com/watch?v=Sr6kJiSzzyc

**Comment by kingqueen3065** [comment]
I now get 502 errors on classifying requests as successful. One of our users (who is not an Admin or a Pro Admin) also experiences this, e.g. they did with https://www.whatdotheyknow.com/request/how_many_foi_requests_have_been_2

**Comment by lizconlan** [comment]
hmm 502 Bad Gateway :(

**Comment by lizconlan** [comment]
Huh, I just got it as well on the live site. Cannot reproduce locally

**Comment by lizconlan** [comment]
oh

```
2017/05/17 15:11:45 [error] 5043#5043: *1055266 upstream sent too big header while reading response header from upstream, client: [my_ip], server: www.whatdotheyknow.com, request: "POST /request/271993/describe HTTP/1.1", upstream: "http://[local_ip]/request/271993/describe", host: "www.whatdotheyknow.com", referrer: "https://www.whatdotheyknow.com/request/developer_test"
```

**Comment by lizconlan** [comment]
Part of #4007

**Comment by sagepe** [comment]
Looking back through the logs there seems to have been an uptick in the number of these errors recorded from around mid/late March. Prior to this there were very few, modulus the occasional spike.

When proxying though Nginx, response headers need to fit inside whatever [`proxy_buffer_size`](http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_buffer_size) is set to, even when [`proxy_buffering`](http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_buffering) is disabled (as it is in our case). Increasing the size of this buffer to 8k seems to have resolved this problem. [A common reason to increase this is to cater for large cookies](https://forum.nginx.org/read.php?2,247230,247246#msg-247246) and I understand that there may have been an increase in the size of the session cookie that could account for this (@lizconlan can comment on this).

**Comment by lizconlan** [comment]
From tinkering on my local dev server, it looks as though the session cookie string alone can get to 3.5k - we'll try to fix that (hopefully #3978 might help) - so [@sagepe's comment above](https://github.com/mysociety/alaveteli/issues/3922#issuecomment-304884080) seems spot on

Closing after checking that classifying a request as successful no longer triggers the error. Please reopen if it reoccurs
